### PR TITLE
refactor(frontend): split agent studio session actions

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
- Split the oversized Agent Studio session action facade into focused send, question, selection, and shared state action modules.
- Kept the public `useAgentStudioSessionActions` facade stable for the orchestration controller while removing unused bootstrap/hydration inputs.
- Grouped extracted internals under `pages/agents/session-actions/` with local codemap documentation and focused send-action coverage.

## Goal
This PR reduces coupling in Agent Studio session action handling. The previous hook mixed send guards, session start/reuse, question submission, selection routing, activity state, and UI feedback in one large body, which made edge cases harder to test and reason about.

## Technical approach
- Extracted send behavior into `useAgentStudioSendAction`, preserving guard order, reusable prompt expansion, attachment validation/staging, start/reuse behavior, and async activity cleanup.
- Extracted question-answer and selection handling into dedicated hooks with purpose-specific inputs.
- Derived base active-session state once, then composed send-aware `isSessionWorking` in the facade so create/selection/stop state stays consistent during in-flight sends.
- Kept queued-message support descriptor-driven and avoided backend, contract, runtime schema, or persistence changes.
- Added focused tests for the extracted send action plus selection/question boundary coverage while retaining facade compatibility tests.
- Moved extracted session action internals into `packages/frontend/src/pages/agents/session-actions/` to keep the crowded Agent Studio route folder easier to navigate.

## Verification
- `bun run lint`
- `bun run typecheck`
- `bun run test`
- `bun run build`
- `bun run check:rust`
- `bun run test:rust`
- `cd apps/desktop/src-tauri && cargo fmt --all --check`
- `cd apps/desktop/src-tauri && cargo clippy --workspace --all-targets -- -D warnings`
- `cd apps/desktop/src-tauri && cargo build`

## Summary by CodeRabbit

* **Documentation**
  * Added detailed docs describing Agent Studio session-action internals and updated integration points.

* **Refactor**
  * Reorganized session management into smaller, focused hooks for clearer behavior and maintainability.

* **Bug Fixes / UX**
  * Improved send/start flow: prevents duplicate session starts, blocks inappropriate concurrent sends, and better handles attachment staging errors.

* **Tests**
  * Expanded tests covering send behavior, selection handlers, and question-submission flows.

### CODE REVIEW

File: packages/frontend/src/pages/agents/session-actions/use-agent-studio-question-actions.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  );
+
+  useEffect(() => {
+    setIsSubmittingQuestionByRequestId((current) => {
+      if (activeExternalSessionId === null && Object.keys(current).length === 0) {
+        return current;
+      }
+      return {};
+    });
+  }, [activeExternalSessionId]);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/agents/session-actions/use-agent-studio-question-actions.ts]: Fixed by tracking question submission state per active session instead of clearing it on session changes.

File: packages/frontend/src/pages/agents/use-agent-studio-session-actions.ts [hunk 2]
```diff
@@ -218,7 +164,7 @@ export function useAgentStudioSessionActions({
     selectedTask,
     agentStudioReady,
     isActiveTaskHydrated,
-    isSessionWorking,
+    isSessionWorking: startFlowSessionWorking,
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/agents/use-agent-studio-session-actions.ts:167]: Fixed by blocking quick actions while the session is working/sending and adding a synchronous start-key guard for duplicate starts.

File: packages/frontend/src/pages/agents/use-agent-studio-session-actions.ts [hunk 3]
```diff
@@ -230,358 +176,77 @@ export function useAgentStudioSessionActions({
...
-      selectedTask,
-      selectedModelDescriptor?.attachmentSupport,
-      sendAgentMessage,
-      startSession,
-      taskId,
-    ],
-  );
+  const isSessionWorking =
+    sessionState.hasActiveSession && (sessionState.isSessionBusy || isSending);
+  const busySendBlockedReason = getBusySendBlockedReason(isSessionWorking);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/agents/use-agent-studio-session-actions.ts]: Fixed by aligning the send hook input and facade busy reason so they use the same status-based queued-message block condition.

**Comment by Maxsky5** [comment]
Follow-up: addressed the Greptile summary items by adding the explicit `useAgentStudioSessionActionState` return type and aligning the busy-send block condition. The CodeRabbit rate-limit comment is non-actionable, so no code change is needed for that notice.

File: packages/frontend/src/pages/agents/session-actions/use-agent-studio-send-action.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        sendContextKeys.add(activeComposerContextKey);
+        sendContextKeys.add(targetComposerContextKey);
+
+        setSendingActivityCountByContext((current) => {
+          let next = current;
+          for (const contextKey of sendContextKeys) {
+            next = incrementActivityCountRecord(next, contextKey);
+          }
+          return next;
+        });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/agents/session-actions/use-agent-studio-send-action.ts]: Fixed by adding a synchronous ref gate and incrementing the active send context before `startSession()` can suspend. Added a focused double-send test and verified `deps:check` now passes with `fast-uri@3.1.2`.
